server/src/main/java/org/opensearch/client/AdminClient.java

```diff
@@ -32,13 +32,16 @@
 
 package org.opensearch.client;
 
+import org.opensearch.common.annotation.PublicApi;
+
 /**
  * Administrative actions/operations against the cluster or the indices.
  *
  * @see org.opensearch.client.Client#admin()
  *
- * @opensearch.internal
+ * @opensearch.api
  */
+@PublicApi(since = "1.0.0")
```

Comment:
Ooh.. I think we should keep server classes internal until they find their JPMS home? This will likely change namespace. Or is the annotation check just making sure class cosmetics don't change and not enforcing namespace changes?

Reply:
> Ooh.. I think we should keep server classes internal until they find their JPMS home? This will likely change namespace. Or is the annotation check just making sure class cosmetics don't change and not enforcing namespace changes?

So the `Plugin` has this hoot to allow plugin developers create own components, the `Client -> AdminClient`  is part of it, we could pretend it is internal but it is not :(

```
    public Collection<Object> createComponents(
        Client client,
        ClusterService clusterService,
        ThreadPool threadPool,
        ResourceWatcherService resourceWatcherService,
        ScriptService scriptService,
        NamedXContentRegistry xContentRegistry,
        Environment environment,
        NodeEnvironment nodeEnvironment,
        NamedWriteableRegistry namedWriteableRegistry,
        IndexNameExpressionResolver indexNameExpressionResolver,
        Supplier<RepositoriesService> repositoriesServiceSupplier
    ) {
        return Collections.emptyList();
    }

``